### ISSUE TITLE
Shell force / moment variables not written in vtk output

### ISSUE THREAD

**Issue by clazaro** [KratosMultiphysics/Kratos#10506]
**Description**
I am trying to visualize shell internal forces/moments with ParaView. I can export Von Mises stresses by including for example `"VON_MISES_STRESS_TOP_SURFACE"` in the `"gauss_point_variables_extrapolated_to_nodes"` dictionary entry of `"vtk_output"` in the `ProjectParameters.json`. However, if I include `"SHELL_FORCES"`, `"SHELL_MOMENTS"` or the global versions, no warning is thrown, but no results are generated for these variables. This problem happens just with vtk output, not with gid output.

**Scope**
Kratos/input_output/vtk_output

**To Reproduce**
This is the vtk_output block:
```
        "vtk_output" : [{
            "python_module" : "vtk_output_process",
            "kratos_module" : "KratosMultiphysics",
            "process_name"  : "VtkOutputProcess",
            "help"          : "This process writes postprocessing files for Paraview",
            "Parameters"    : {
                "model_part_name"                             : "Structure",
                "output_control_type"                         : "step",
                "output_interval"                             : 1,
                "file_format"                                 : "ascii",
                "output_precision"                            : 7,
                "output_sub_model_parts"                      : false,
                "output_path"                                 : "vtk_output",
                "save_output_files_in_folder"                 : true,
                "nodal_solution_step_data_variables"          : [
                    "DISPLACEMENT",
                    "REACTION",
                    "ROTATION"
                ],
                "nodal_data_value_variables"                  : [],
                "element_data_value_variables"                : [],
                "condition_data_value_variables"              : [],
                "gauss_point_variables_extrapolated_to_nodes" : [
                    "VON_MISES_STRESS_TOP_SURFACE",
                    "VON_MISES_STRESS_MIDDLE_SURFACE",
                    "VON_MISES_STRESS_BOTTOM_SURFACE",
                    "SHELL_FORCE_GLOBAL",
                    "SHELL_MOMENT_GLOBAL"
                ]
            }
        }]
```

**Expected behavior**
Shell force / moment blocks to be generated in the vtk file, or if this is not implemented, a warning message should be thrown.

**Environment**
Please specify your operating system, kratos branch, python version or any other library versions that you consider relevant.
E.g.:
- OS: Ubuntu 22.04
- Branch: master
- Python 3.10

**Comment by loumalouomega** [comment]
I cannot check right now, but the variables that are compatible with VTK are the following:

https://github.com/KratosMultiphysics/Kratos/blob/a88ba3975f66dbecbbd2fd738fe6c1d4c0dd017b/kratos/input_output/vtk_output.cpp#L435

The Variable<Matrix> is not compatible, and SHELL_FORCE_GLOBAL and  SHELL_MOMENT_GLOBAL are both matrix. We would need to implement it as tensor: https://kitware.github.io/vtk-examples/site/VTKFileFormats/

**Comment by loumalouomega** [comment]
> I cannot check right now, but the variables that are compatible with VTK are the following:
> 
> https://github.com/KratosMultiphysics/Kratos/blob/a88ba3975f66dbecbbd2fd738fe6c1d4c0dd017b/kratos/input_output/vtk_output.cpp#L435
> 
> The Variable is not compatible, and SHELL_FORCE_GLOBAL and SHELL_MOMENT_GLOBAL are both matrix. We would need to implement it as tensor: [kitware.github.io/vtk-examples/site/VTKFileFormats](https://kitware.github.io/vtk-examples/site/VTKFileFormats/)

(In our IO tensor is not currently available)

**Comment by clazaro** [comment]
An additional comment:
Would it be possible to convert the variables to Voigt type and then export? When looking at the results, one is normally interested in the single components (membrane forces, bending moments etc.) not in the tensor character.

**Comment by loumalouomega** [comment]
> I see, thanks. However, it is not throwing any warning as one should expect, according to
> 
> https://github.com/KratosMultiphysics/Kratos/blob/a88ba3975f66dbecbbd2fd738fe6c1d4c0dd017b/kratos/input_output/vtk_output.cpp#L688

Yes, that is strange...

**Comment by loumalouomega** [comment]
> An additional comment: Would it be possible to convert the variables to Voigt type and then export? When looking at the results, one is normally interested in the single components (membrane forces, bending moments etc.) not in the tensor character.

What do you mean? do you want to have each component indecently?

**Comment by clazaro** [comment]
Yes, I want to represent the value of each single component to get a map of, say, axial forces or twisting moments (and possibly their principal values)

**Comment by loumalouomega** [comment]
> Yes, I want to represent the value of each single component to get a map of, say, axial forces or twisting moments (and possibly their principal values)

In paraview you can select the component of a solution (default is magnitude)

**Comment by clazaro** [comment]
Yes, if the variable is a vector. But as you pointed out, SHELL_FORCES and SHELL_MOMENTS are not, and therefore can't be exported. This is why I was asking if it is possible to transform them to Voigt vectors, which would allow the export.

**Comment by loumalouomega** [comment]
> Yes, if the variable is a vector. But as you pointed out, SHELL_FORCES and SHELL_MOMENTS are not, and therefore can't be exported. This is why I was asking if it is possible to transform them to Voigt vectors, which would allow the export.

It is possible, but as I don't know the details about the implementation, I cannot do it. Right now working on giving support to tensors in VTK, that will work too

**Comment by clazaro** [comment]
Thanks!

**Comment by loumalouomega** [comment]
I would close once is implemented, I cannot dedicate 100% to this
